src/nvim/screen.c

```diff
@@ -2653,20 +2663,6 @@ win_line (
       cur = cur->next;
   }
 
-  /* Cursor line highlighting for 'cursorline' in the current window.  Not
-   * when Visual mode is active, because it's not clear what is selected
-   * then. */
-  if (wp->w_p_cul && lnum == wp->w_cursor.lnum
-      && !(wp == curwin && VIsual_active)) {
-    if (line_attr != 0 && !(State & INSERT) && bt_quickfix(wp->w_buffer)
-        && qf_current_entry(wp) == lnum) {
-      line_attr = hl_combine_attr(win_hl_attr(wp, HLF_CUL), line_attr);
```

Comment:
Why isn't this branch needed anymore?

Reply:
The old one basically says if we are in quickfix window, then we use `HLF_CUL` to complement current `line_attr`, otherwise it will directly set `line_attr` to `HLF_CUL`.  Setting `line_attr` to `HLF_CUL` is not what we want, since it incorrectly override some highlights under some condition (the logic is complex here)....


In this patch, I split `line_attr` into `line_attr` and `line_attr_low_priority`.  Now `line_attr_low_priority` only takes care of `cursorline`, and it will be merged into `char_attr` with lowest priority.  For other `line_attr`, it will still use the old logic.  Since `line_attr_low_priority` already has lowest priority, I believe these lines are not necessary.